docs-user/bunny.md

```diff
@@ -0,0 +1,272 @@
+
+The following article is a case study in using the profiler to identify a performance issue, and then fixing it for a 4 times performance improvement. This improvement moves the frame rate from a fairly slow 15fps to a smooth 60fps.
+
+## The project description
+
+![A picture of a 3d bunny rabbit model rendered using small squares.](./images/bunny-analysis/bunny.png)
+
+The project is a website that takes user's JavaScript code, and runs it to produce a visualization. The user's code only has access to the function `rect(color, x, y, width, height)`, which draws a rectangle to the screen. For the implementation, the website posts the user's code to a sandboxed iframe. The iframe has a `<canvas>` element that is rendered to via the [`CanvasRenderingContext2D`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D) API. The user's code is evaluated in a WebWorker, and then the results of `rect()` are posted back to the iframe's code.
+
+In a simplified example, the `worker.js` would run something like:
+
+```js
+// Evaluate the user's code string:
+eval(`
+rect('#fff', 10, 10, 3, 3);
+rect('#fff', 20, 20, 3, 3);
+rect('#fff', 30, 30, 3, 3);
+rect('#fff', 40, 40, 3, 3);
+rect('#fff', 50, 50, 3, 3);
+`);
+
+// worker.js would post something like this to the iframe:
+self.postMessage({
+  color: ['#fff', '#fff', '#fff', '#fff', '#fff'],
+  x: [10, 20, 30, 40, 50],
+  y: [10, 20, 30, 40, 50],
+  w: [3, 3, 3, 3, 3],
+  h: [3, 3, 3, 3, 3],
+});
+```
+
+Then the iframe would draw the code using:
+
+```js
+worker.addEventListener('message', message => {
+  const { data } = message;
+  for (let i = 0; i < data.color.length; i++) {
+    ctx.fillStyle = data.color[i];
+    ctx.fillRect(data.x[i], data.y[i], data.w[i], data.h[i]);
+  }
+});
+```
+
+This would result in user's evaluated code drawing something to the screen, like the bunny model in the above image.
+
+## The problem
+
+This code ended up not scaling well for large sets of rectangles being drawn to the screen. There were lots of stutters, and a slow frame rate. The user impact for fixing this problem would be to have a smoother frame rate, plus the ability to draw many more rectangles to the screen without slowing things down. In order to validate fixes, the following steps were used to reproduce the issue.
+
+ * Load the page with the bunny visualization.
+ * Hit Ctrl Shift 1 to turn on the Gecko Profiler.
+ * Wait around 5 seconds
+ * Hit Ctrl Shift 2 to capture the profile.
+ * Set the range to 3.0 seconds of relatively stable frames that don't have stutters or GC pauses.
+ * Hide idle stacks by right clicking `__psync_cvwait` and `mach_msg_trap` in the Flame Graph, and choosing "Drop samples with this function".
+ * Filter the threads to:
+   * The relevant content process
+   * The relevant DOM Worker
+   * The compositor
+
+This resulted in the following profile for the baseline profile: [https://perfht.ml/2IxTwqi](https://perfht.ml/2IxTwqi)
```

Comment:
I'd put the URL before the STR. Something like "Here is a profile that exhibits the problem we try to resolve: XXX. To capture and prepare it, here are the steps that were realized:".

Also (optional nit) in markdown you can put the URL directly and it should be automatically converted, you don't need to use the URL syntax. Same comment for other URLs below of course.